subsys/shell/shell_cmds.c

```diff
@@ -394,94 +348,70 @@ static int cmd_history(const struct shell *shell, size_t argc, char **argv)
 static int cmd_shell_stats_show(const struct shell *shell, size_t argc,
 				char **argv)
 {
+	ARG_UNUSED(argc);
 	ARG_UNUSED(argv);
 
 	if (!IS_ENABLED(CONFIG_SHELL_STATS)) {
-		shell_fprintf(shell, SHELL_ERROR, "Command not supported.\n");
+		shell_error(shell, "Command not supported.");
 		return -ENOEXEC;
 	}
 
-	int ret = shell_cmd_precheck(shell, (argc == 1), NULL, 0);
-
-	if (ret == 0) {
-		shell_fprintf(shell, SHELL_NORMAL, "Lost logs: %u\n",
-			      shell->stats->log_lost_cnt);
-	}
+	shell_print(shell, "Lost logs: %u", shell->stats->log_lost_cnt);
 
-	return ret;
+	return 0;
 }
 
 static int cmd_shell_stats_reset(const struct shell *shell,
 				 size_t argc, char **argv)
 {
+	ARG_UNUSED(argc);
 	ARG_UNUSED(argv);
 
 	if (!IS_ENABLED(CONFIG_SHELL_STATS)) {
-		shell_fprintf(shell, SHELL_ERROR, "Command not supported.\n");
+		shell_error(shell, "Command not supported.");
 		return -ENOEXEC;
 	}
 
-	int ret = shell_cmd_precheck(shell, (argc == 1), NULL, 0);
-
-	if (ret == 0) {
-		shell->stats->log_lost_cnt = 0;
-	}
+	shell->stats->log_lost_cnt = 0;
 
-	return ret;
+	return 0;
 }
 
 static int cmd_shell_stats(const struct shell *shell, size_t argc, char **argv)
 {
 	ARG_UNUSED(argc);
 
-	if (shell_help_requested(shell)) {
-		shell_help_print(shell, NULL, 0);
-		return 1;
-	} else if (argc == 1) {
-		shell_help_print(shell, NULL, 0);
-	} else {
-		shell_fprintf(shell, SHELL_ERROR, "%s:%s%s\n", argv[0],
-			      SHELL_MSG_UNKNOWN_PARAMETER, argv[1]);
-	}
+	shell_print(shell, "%s:%s%s", argv[0],
+		    SHELL_MSG_UNKNOWN_PARAMETER, argv[1]);
 
 	return -EINVAL;
 }
 
 static int cmd_resize_default(const struct shell *shell,
 			      size_t argc, char **argv)
 {
+	ARG_UNUSED(argc);
 	ARG_UNUSED(argv);
 
-	int ret = shell_cmd_precheck(shell, (argc == 1), NULL, 0);
-
-	if (ret == 0) {
-		SHELL_VT100_CMD(shell, SHELL_VT100_SETCOL_80);
-		shell->ctx->vt100_ctx.cons.terminal_wid =
-						   SHELL_DEFAULT_TERMINAL_WIDTH;
-		shell->ctx->vt100_ctx.cons.terminal_hei =
-						  SHELL_DEFAULT_TERMINAL_HEIGHT;
-	}
+	SHELL_VT100_CMD(shell, SHELL_VT100_SETCOL_80);
+	shell->ctx->vt100_ctx.cons.terminal_wid =  SHELL_DEFAULT_TERMINAL_WIDTH;
+	shell->ctx->vt100_ctx.cons.terminal_hei = SHELL_DEFAULT_TERMINAL_HEIGHT;
 
-	return ret;
+	return 0;
 }
 
 static int cmd_resize(const struct shell *shell, size_t argc, char **argv)
 {
 	int err;
 
 	if (!IS_ENABLED(CONFIG_SHELL_CMDS_RESIZE)) {
-		shell_fprintf(shell, SHELL_ERROR, "Command not supported.\n");
+		shell_error(shell, "Command not supported.");
 		return -ENOEXEC;
 	}
 
-	err = shell_cmd_precheck(shell, (argc <= 2), NULL, 0);
-	if (err) {
-		return err;
-	}
-
 	if (argc != 1) {
```

Comment:
this check here looks strange. What about second optional parameter (size?). I don't see it handled anywhere.

Reply:
Command `resize` be called with 1 argument `default`. This check prevents from typo mistake in `resize default`.